pkg/node/node.go

```diff
@@ -93,7 +93,26 @@ func NewNodeServer(ctx context.Context, identity, nodeID, rack, zone, region str
 		setQuota:        xfs.SetQuota,
 	}
 
+	handler := &ueventHandler{
+		nodeID:          nodeID,
+		directCSIClient: client.GetDirectClientset(),
+		topology: map[string]string{
+			string(utils.TopologyDriverIdentity): identity,
+			string(utils.TopologyDriverRack):     rack,
+			string(utils.TopologyDriverZone):     zone,
+			string(utils.TopologyDriverRegion):   region,
+			string(utils.TopologyDriverNode):     nodeID,
+		},
+		dynamicDriveDiscovery: dynamicDriveDiscovery,
+	}
+	if dynamicDriveDiscovery {
+		klog.V(3).Info("Doing initial drive sync up")
+		handler.syncDrives(ctx)
```

Comment:
shouldn't the initial sync be happening even when `dynamicDriveDiscovery` is not enabled? We can  bring this out of the loop.